lib/cc/cli/init.rb

```diff
@@ -10,7 +10,7 @@ class Init < Command
       def run
         if !upgrade? && filesystem.exist?(CODECLIMATE_YAML)
           warn "Config file .codeclimate.yml already present.\nTry running 'validate-config' to check configuration."
-          create_default_configs
+          create_default_engine_configs
```

Comment:
Is there a strong reason for these renames (inserting `_engine_`)? I'm finding it distracting when trying to make out the actual code change here.

Given your PR description I expected a smaller change, something along the lines of:

- extract something like `#engine_names` that uses cc.yml if present or `ConfigGenerator` if not
- Replace `config_generator.eligible_engines` with `#engine_names`

The other renames and refactors might be good, but I think it'd be easier to discern if the behavior-changing and non-behavior-changing stuff were in separate PRs.